presto-main/src/test/java/com/facebook/presto/sql/planner/TestPredicatePushdownWithDynamicFilter.java

```diff
@@ -47,14 +47,14 @@ public void testNonStraddlingJoinExpression()
                 anyTree(
                         join(
                                 INNER,
-                                ImmutableList.of(equiJoinClause("ORDERS_OK", "LINEITEM_OK")),
-                                anyTree(tableScan("orders", ImmutableMap.of("ORDERS_OK", "orderkey"))),
+                                ImmutableList.of(equiJoinClause("LINEITEM_OK", "ORDERS_OK")),
                                 anyTree(
-                                        filter(
-                                                "cast('2' as varchar) = cast(LINEITEM_LINENUMBER as varchar)",
+                                        node(
+                                                FilterNode.class,
```

Comment:
why did this change from specifying the filter?

Reply:
Dynamic filter will add a new predicate. It's a lot of work to make the filter matcher to be able to handle it. Existing dynamic filter tests are also just checking for FilterNode.